listener/listener.py

```diff
@@ -0,0 +1,234 @@
+from asyncpg.pool import Pool
+
+from common.logging import init_logging, get_logger
+from common.status_app import create_status_app, create_status_runner
+from common.utils import a_ensure_minimal_schema_version, validate_kafka_msg, send_msg_to_payload_tracker
+from common.mqueue import MQReader, MQWriter
+from common.identity import get_identity, is_entitled_insights
+from common.database_handler import setup_db_pool
+from .common import (CFG, PROMETHEUS_PORT, SKIPPED_MESSAGES, INVALID_IDENTITY, MISSING_INSIGHTS_ENTITLEMENT,
+                     MESSAGE_PARSE_ERROR, InventoryMsgType)
+from .queue import ListenerQueue
+
+LOGGER = get_logger(__name__)
+
+
+class InventoryMsgValidator:
+    """Inventory kafka message validator"""
+
+    _created_updated_msg_fields = {
+        "host": ["id", "org_id", "system_profile", "display_name", ("insights_id", False)],
+        "timestamp": [],
+        "type": [],
+    }
+
+    _upload_msg_fields = {
+        "platform_metadata": ["b64_identity", "url"],
+    }
+
+    _delete_msg_fields = {
+        "id": [],
+        "org_id": [],
+        "type": [],
+        "insights_id": [],
+    }
+
+    def __init__(self, payload_tracker: MQWriter):
+        """Constructor"""
+        self.payload_tracker = payload_tracker
+
+    def _validate_created_updated(self, msg: dict) -> (bool, str):
+        """Validates CREATED/UPDATED kafka message"""
+        if not validate_kafka_msg(msg, self._created_updated_msg_fields):
+            SKIPPED_MESSAGES.inc()
+            return False, None
+        if msg.get("platform_metadata", None):
+            if not validate_kafka_msg(msg, self._upload_msg_fields):
+                SKIPPED_MESSAGES.inc()
+                return False, None
+            LOGGER.info("Received created/updated msg, inventory_id: %s, type: %s", msg["host"]["id"], msg["type"])
+            send_msg_to_payload_tracker(self.payload_tracker, msg, "received")
+            identity = get_identity(msg["platform_metadata"]["b64_identity"])
+            if identity is None:
+                INVALID_IDENTITY.inc()
+                error_msg = "Skipped upload due to invalid identity header."
+                LOGGER.warning(error_msg)
+                return False, error_msg
+            if not is_entitled_insights(identity, allow_missing_section=True):
+                MISSING_INSIGHTS_ENTITLEMENT.inc()
+                error_msg = "Skipped upload due to missing insights entitlement."
+                LOGGER.debug(error_msg)
+                return False, error_msg
+        else:
+            LOGGER.info("Received update event msg, inventory_id: %s", msg["host"]["id"])
+        return True, None
+
+    def _validate_deleted(self, msg: dict) -> (bool, str):
+        """Validates DELETE inventory kafka message"""
+        if not validate_kafka_msg(msg, self._delete_msg_fields):
+            SKIPPED_MESSAGES.inc()
+            return False, None
+        return True, None
```

Comment:
second return value is always `None`, I guess you created it to make it consistent with `_validate_created_updated`, so at least add `Optional[str]` as return type